Binary files wmconfig-0.9.10.orig/.fvwm2.c.swp and wmconfig-0.9.10/.fvwm2.c.swp differ
diff -urN wmconfig-0.9.10.orig/fvwm2.c wmconfig-0.9.10/fvwm2.c
--- wmconfig-0.9.10.orig/fvwm2.c	Fri Oct 12 17:10:27 2001
+++ wmconfig-0.9.10/fvwm2.c	Fri Oct 12 18:46:24 2001
@@ -36,7 +36,7 @@
 /*
  * Outputs on stdout the FVWM2-compatible configuration
  */
-static void fvwm2(struct group *root, int level, const char *menu_name)
+static void fvwm2(const char *absolute_root_name, struct group *root, int level, const char *menu_name)
 {
     struct item *item;
     int debug_icon = 0;
@@ -90,6 +90,8 @@
 	    printf("AddToMenu\t%s%s\t\"%s\"\tPopup\t%s%s.%s\n",
 		   menu_name, root_name, unicode_to_locale(tmp->name),
 		   menu_name, root_name, c_tmp);
+	    printf("AddToFunc\tDestroy%sMenu\tI DestroyMenu \"%s%s.%s\"\n",
+		   absolute_root_name, menu_name, root_name, c_tmp);
 	    free(c_tmp);
 	} else if (item->type == ITEM_APP) {
 	    struct package *app;
@@ -99,10 +101,10 @@
 		   app->__package?app->__package:"");
 	    if (app->name && app->exec) {
 		if ((app->mini_icon != NULL) && !is_set(NO_MINI_ICONS))
-		    printf("AddToMenu\t%s%s\t\"%s %%%s%%\"\tExec\t%s\n",
+		    printf("AddToMenu\t%s%s\t\"%s %%%s%%\"\tExec exec\t%s\n",
 			   menu_name, root_name, unicode_to_locale(app->name), app->mini_icon, app->exec);
 		else
-		    printf("AddToMenu\t%s%s\t\"%s\"\tExec\t%s\n",
+		    printf("AddToMenu\t%s%s\t\"%s\"\tExec exec\t%s\n",
 			   menu_name, root_name, unicode_to_locale(app->name), app->exec);
 	    }
 	}
@@ -116,7 +118,7 @@
 
 	    c_tmp = malloc(BUFSIZ);
 	    snprintf(c_tmp, BUFSIZ, "%s%s.", menu_name, root_name);
-	    fvwm2(item->data, level+1, c_tmp);
+	    fvwm2(absolute_root_name, item->data, level+1, c_tmp);
 	    free(c_tmp);
 	}
 	item++;
@@ -126,7 +128,9 @@
 
 void output_fvwm2(struct group *root)
 {
-    fvwm2(root, 0, "");
+    char *absolute_root_name;
+    absolute_root_name = single_string(unicode_to_locale(root->name));
+    fvwm2(absolute_root_name, root, 0, "");
 }
 
      
diff -urN wmconfig-0.9.10.orig/wmconfig.1 wmconfig-0.9.10/wmconfig.1
--- wmconfig-0.9.10.orig/wmconfig.1	Mon Jul 10 21:43:34 2000
+++ wmconfig-0.9.10/wmconfig.1	Fri Oct 12 18:44:06 2001
@@ -1,4 +1,18 @@
 .\"	$Id$	
+.de EX          \"Begin example (stolen from fvwm2.1)
+.ne 5
+.if n .sp 1
+.if t .sp .5
+.nf
+.in +.5i
+..
+.de EE		\"End example (stolen from fvwm2.1)
+.fi
+.in -.5i
+.if n .sp 1
+.if t .sp .5
+..
+
 .TH WMCONFIG 1 "Sun Mar 28 1999"
 .UC 4
 .SH NAME
@@ -108,7 +122,7 @@
 
 .SH OPTIONS
 .IP \fB--output=\fImanager\fR
-Where \fImanager\fR can currently be one of: \fBfvwm2\fR,
+Where \fImanager\fR can currently be one of: \fBfvwm2\fR (check \fBFVWM2 NOTE\fR below),
 \fBfvwm95\fR, \fBmwm\fR, \fBafterstep\fR, \fBicewm\fR, \fBwmaker\fR, \fBkde\fR or \fBdebug\fR. The only
 entry which is special in this list is the \fBdebug\fR entry, which
 will produce a nice tree-like output to debug your settings and show
@@ -210,6 +224,30 @@
 
 .IP \fB--version\fR
 This is the popular "what version was that ?" switch.
+
+.SH FVWM2 NOTE
+\fBfvwm2\fR has ability to dynamically generate programs menu (as \fBGNOME\fR and \fBKDE\fR do). \fBwmconfig\fR takes advantage of this feature. This requires menu file of \fBfvwm2\fR to contain some special magic. If you named menu for programs "Programs" you should add following code to your menu file:
+
+.EX
+AddToMenu Programs
++ DynamicPopupAction Function MakeProgramsMenu
++ DynamicPopdownAction Function DestroyProgramsMenu
+
+AddToFunc MakeProgramsMenu
++ I DestroyMenu recreate Programs
++ I PipeRead 'wmconfig --output fvwm2 --rootmenu Programs'
+.EE
+
+Above code assumes that you have "Programs" menu declaration similar to that:
+
+.EX
+AddToMenu Hotlist "Root menu" Title
+[...]
++ "Programs" Popup Programs
+[...]
+.EE
+
+If you use different name for programs menu you should update above code \fIincluding\fR name of "DestroyProgramsMenu" function (if you use name "Applications" this function will be named "DestroyApplicationsMenu".
  
 .SH BUGS
 .SS Definite bugs
@@ -229,4 +267,7 @@
 Cristian Gafton <gafton@redhat.com>
 Red Hat Software, Inc.
 
+Various patches and updates by PLD Team
+http://www.pld.org.pl/
+
 .fi
